### PR TITLE
Add derivedClass in PageViewModel

### PR DESCRIPTION
**Pull request by qinezh**
@vwxyzh @vicancy @superyyrrzz @hellosnow @ansyral @chenkennt

### CODE REVIEW

File: src/Microsoft.DocAsCode.DataContracts.ManagedReference/ItemViewModel.cs [hunk 1]
```diff
@@ -279,6 +279,11 @@ public string FullNameForVB
         [JsonProperty("inheritance")]
         public List<string> Inheritance { get; set; }
 
+        [YamlMember(Alias = "derivedClass")]
+        [MergeOption(MergeOption.Ignore)]
+        [JsonProperty("derivedClass")]
+        public List<string> DerivedClass { get; set; }
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.DataContracts.ManagedReference/ItemViewModel.cs]: >DerivedClass [](start = 28, length = 12)

DerivedClasses

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Resolvers/SetDerivedClass.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            InitDerivedClassMapping(yaml.Members);
+            AppendDerivedClass(yaml.Members);
+        }
+
+        private void InitDerivedClassMapping(List<MetadataItem> items)
+        {
+            foreach (var item in items ?? Enumerable.Empty<MetadataItem>())
+            {
+                var inheritance = item.Inheritance;
+                if (inheritance?.Any() == true)
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Resolvers/SetDerivedClass.cs]: >Any [](start = 33, length = 3)

use Count for List

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Resolvers/SetDerivedClass.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                {
+                    List<string> derivedClass;
+                    if (_derivedClassMapping.TryGetValue(inheritance.Last(), out derivedClass))
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Resolvers/SetDerivedClass.cs]: >inheritance.Last(), [](start = 57, length = 19)

k[k.Count -1]

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Resolvers/SetDerivedClass.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    using System.Collections.Generic;
+
+    using Microsoft.DocAsCode.DataContracts.ManagedReference;
+
+    public class SetDerivedClass: IResolverPipeline
+    {
+        private readonly Dictionary<string, List<string>> _derivedClassMapping = new Dictionary<string, List<string>>();
+
+        public void Run(MetadataModel yaml, ResolverContext context)
+        {
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Resolvers/SetDerivedClass.cs]: >{ [](start = 8, length = 1)

if members == null || members.Count == 0 return
to make logic below clearer?

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Resolvers/SetDerivedClass.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        private void AppendDerivedClass(List<MetadataItem> items)
+        {
+            foreach (var item in items ?? Enumerable.Empty<MetadataItem>())
+            {
+                if (item.Type == MemberType.Class)
+                {
+                    List<string> derivedClass;
+                    if (_derivedClassMapping.TryGetValue(item.Name, out derivedClass))
+                    {
+                        item.DerivedClass = derivedClass;
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Resolvers/SetDerivedClass.cs]: sort?

**Comment by hellosnow** [comment]
:shipit:

**Comment by ansyral** [comment]
:shipit:

**Comment by vicancy** [comment]
:shipit:

**Pull request merged**